crates/bevy_ui/src/layout_components.rs

```diff
@@ -0,0 +1,437 @@
+            },
+            ..Self::DEFAULT
+        }
+    }
+
+    /// Sets only the padding
+    pub const fn padding(rect: UiRect<Val>) -> Spacing {
+        Spacing {
+            padding: rect,
+            ..Self::DEFAULT
+        }
+    }
+
+    /// Sets the padding on all sides to `val`
+    pub const fn padding_all(val: Val) -> Spacing {
+        Spacing {
+            padding: UiRect {
+                left: val,
+                right: val,
+                bottom: val,
+                top: val,
+            },
+            ..Self::DEFAULT
+        }
+    }
+
+    /// Sets only the padding
+    pub const fn border(rect: UiRect<Val>) -> Spacing {
+        Spacing {
+            border: rect,
+            ..Self::DEFAULT
+        }
+    }
+
+    /// Sets the border on all sides to `val`
+    pub const fn border_all(val: Val) -> Spacing {
+        Spacing {
+            border: UiRect {
+                left: val,
+                right: val,
+                bottom: val,
+                top: val,
+            },
+            ..Self::DEFAULT
+        }
+    }
+}
+
+/// Defines the text direction
+///
+/// For example English is written LTR (left-to-right) while Arabic is written RTL (right-to-left).
+#[derive(
+    Component, Copy, Clone, PartialEq, Eq, Debug, Default, Serialize, Deserialize, Reflect,
+)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum TextDirection {
+    /// Inherit from parent node
+    #[default]
+    Inherit,
+    /// Text is written left to right
+    LeftToRight,
+    /// Text is written right to left
+    RightToLeft,
+}
+
+/// Whether to show or hide overflowing items
+#[derive(
+    Component, Copy, Clone, PartialEq, Eq, Debug, Default, Reflect, Serialize, Deserialize,
+)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum Overflow {
+    /// Show overflowing items
+    #[default]
+    Visible,
+    /// Hide overflowing items
+    Hidden,
+}
+
+/// Flexbox-specific layout components
+pub mod flex {
+
+    use bevy_ecs::query::{Changed, Or, WorldQuery};
+
+    use super::*;
+
+    /// A query for all of the components need for flexbox layout.
+    ///
+    /// See [`FlexLayoutChanged`] when attempting to use this as a query filter.
+    #[derive(WorldQuery)]
+    pub struct FlexLayoutQuery {
+        /// The layout algorithm used
+        pub layout_strategy: &'static LayoutStrategy,
+        /// The position of this UI node
+        pub offset: &'static Offset,
+        /// Whether the node should be absolute or relatively positioned
+        pub position_type: &'static PositionType,
+        /// The constraints on the size of this node
+        pub size_constraints: &'static SizeConstraints,
+        /// The margin, padding and border of the UI node
+        pub spacing: &'static Spacing,
+        /// The flexbox layout parameters
+        pub flex_layout: &'static FlexLayout,
+        /// The direction of the text
+        pub text_direction: &'static TextDirection,
```

Comment:
AFAIK `TextDirection` isn't used by flex

Reply:
> AFAIK `TextDirection` isn't used by flex

The `TextDirection` is used to determine what `start` and `end` means, e.g. in LTR `start` will be `left`, in RTL it will be `right`.

More info: <https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Basic_Concepts_of_Flexbox#start_and_end_lines>

Reply:
This is done using `flex-direction` not `TextDirection`, which isn't part of `Flex`. Also, `TextDirection` isn't sent to `Taffy`: 

https://github.com/bevyengine/bevy/blob/b4b7b0bb4be1d7bc6f8553a89eb9daffbef05246/crates/bevy_ui/src/flex/convert.rs#L40

Reply:
It's a combination of both `flex-direction` and `TextDirection`.

The idea is that the layout aligns with the reading direction that users are used to: If I want to make a menu bar where the user should first see the logo, then the name of the app and then navigation links, I would use `flex-direction: row`.
When English text is used, then we have a LTR `TextDirection` and the logo is shown on the left, then the name to the right of that and navigation links right again.
If Arabic text is used, then we have RTL `TextDirection` and the logo is shown on the right, then the name to the left of that and navigation links to the left.

This is a key feature of flexbox, because people who use RTL writing modes will also "read the UI" from right to left. This is why it changes depending on the locale that is currently used.

Reply:
And regarding `taffy`: You are correct, unfortunately `taffy` doesn't support writing modes yet. Hopefully we can add support for that in the future, to align with the Flexbox spec: <https://github.com/DioxusLabs/taffy/issues/213>